avocado_vt/plugins/vt_resolver.py

```diff
@@ -0,0 +1,55 @@
+import warnings
+
+from avocado.core.nrunner import Runnable
+from avocado.core.plugin_interfaces import Resolver
+from avocado.core.resolver import (ReferenceResolution,
+                                   ReferenceResolutionResult)
+from avocado.core.settings import settings
+
+from ..discovery import DiscoveryMixIn
+
+
+class VTResolver(Resolver, DiscoveryMixIn):
+
+    name = 'vt'
+    description = 'Test resolver for Avocado-VT tests'
+
+    def _parameters_to_runnable(self, params):
+        params = self.convert_parameters(params)
+        url = params.get('name')
+        vt_params = params.get('vt_params')
+
+        # Flatten the vt_params, discarding the attributes that are not
+        # scalars, and will not be used in the context of nrunner
+        for key in ('_name_map_file', '_short_name_map_file', 'dep'):
+            if key in vt_params:
+                del(vt_params[key])
+
+        return Runnable('avocado-vt', url, **vt_params)
+
+    def resolve(self, reference):
+        self.config = settings.as_dict()
+        try:
+            cartesian_parser = self._get_parser()
+            self._save_parser_cartesian_config(cartesian_parser)
+        except Exception as details:
+            return ReferenceResolution(reference,
+                                       ReferenceResolutionResult.ERROR,
+                                       info=details)
+
+        cartesian_parser.only_filter(reference)
```

Comment:
This is a very good point, I will investigate further and find a solution.

Reply:
I had to do more investigation and some tests because this is part of PR from @clebergnu. It works as expected and multiple command line filters are working. Because the `cartesian_parser` parser use those, so when you call `cartesian_parser.get_dicts()` the data are already filtered with command line filters.

Reply:
I thought the Cartesian parser parses all variants and the current `only_filter` line restricts these to a reference afterwards. If not then what do you think could be the role of this line? Also how would you pass multiple `only` and `no` restrictions from the avocado command line if you don't mind sharing? Did you try some example?

Reply:
Yes the `only_filter` line restricts these to a reference added by the user and that is desired behavior. So you can run: 
```
avcoado run --test-runner=nrunner --nrunner-max-parallel-tasks 1 type_specific.io-github-autotest-qemu.migrate.default.tcp
JOB ID     : 194061010f2cfb9ec71a98389380de22ec8a5cd2
JOB LOG    : /home/jarichte/avocado/job-results/job-2021-04-21T14.12-1940610/job.log
 (1/3) type_specific.io-github-autotest-qemu.migrate.default.tcp.default: STARTED
 (1/3) type_specific.io-github-autotest-qemu.migrate.default.tcp.default:  PASS (3.08 s)
 (2/3) type_specific.io-github-autotest-qemu.migrate.default.tcp.with_filter_off.with_post_copy: STARTED
 (2/3) type_specific.io-github-autotest-qemu.migrate.default.tcp.with_filter_off.with_post_copy:  PASS (3.11 s)
 (3/3) type_specific.io-github-autotest-qemu.migrate.default.tcp.with_filter_off.with_multifd: STARTED
 (3/3) type_specific.io-github-autotest-qemu.migrate.default.tcp.with_filter_off.with_multifd:  PASS (3.12 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/jarichte/avocado/job-results/job-2021-04-21T14.12-1940610/results.html
JOB TIME   : 11.87 s
```
Which runs all the `type_specific.io-github-autotest-qemu.migrate.default.tcp` tests. And you can restrict those by `only` and `no` restrictions like this:
```
avocado run --test-runner=nrunner --nrunner-max-parallel-tasks=1 --vt-only-filter=with_filter_off --vt-no-filter=with_multifd type_specific.io-github-autotest-qemu.migrate.default.tcp
JOB ID     : 98ba64dab82bd388f3f50d4474ebabb44651e315
JOB LOG    : /home/jarichte/avocado/job-results/job-2021-04-21T14.32-98ba64d/job.log
 (1/1) type_specific.io-github-autotest-qemu.migrate.default.tcp.with_filter_off.with_post_copy: STARTED
 (1/1) type_specific.io-github-autotest-qemu.migrate.default.tcp.with_filter_off.with_post_copy:  PASS (3.12 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/jarichte/avocado/job-results/job-2021-04-21T14.32-98ba64d/results.html
JOB TIME   : 4.41 s
```